client/core/core.go

```diff
@@ -2904,6 +2906,61 @@ func (c *Core) tipChange(assetID uint32, nodeErr error) {
 	c.updateBalances(counts)
 }
 
+func (c *Core) PromptShutdown() bool {
+	c.connMtx.Lock()
+	defer c.connMtx.Unlock()
+	ok := true
+	for _, dc := range c.conns {
+		if dc.hasActiveOrders() {
+			ok = false
+			break
+		}
+	}
+	if !ok {
+		fmt.Print("You have active orders. Shutting down now may result in failed swaps and account penalization. " +
+			"Do you want to quit anyway? ('y' to quit, enter to abort):")
+		scanner := bufio.NewScanner(os.Stdin)
+		scan := make(chan bool)
+		go func() {
+			scan <- scanner.Scan()
+		}()
+		select {
+		case <-c.ctx.Done():
+			return ok
+		case <-scan:
+		}
```

Comment:
This is interesting, as the original reason for this was to listen for ctrl-c, but that send a signal to the `killChan` here and is ignored. If that is still the intent, perhaps this should select accross `killChan` and not `c.ctx`? Otherwise, I don't think it is possible to cancel the main context here.

Also, should return `true` to ensure shutdown?

Reply:
Yeah, good point, @JoeGruffins.  The ctx will only be canceled if a previous `PromptShutdown` allowed `shutdown` to cancel it.

But consider the problem in the following sequence:
- CTRL+C is hit once, receiving from `killChan` and starting the prompt.
- A second CTRL+C is hit while the prompt is still up, which sends to the `killChan` even though there is no longer a receiver since it is buffered.
- The user responds in the negative, causing `shutdown` to start again.
- `shutdown` immediately receives from the `killChan`, and the prompt will be shown again.

Reply:
The second `ctrl+c` would have the same effect as printing `yes` and the user would not be able to respond in the negative afterwards, I think. It looks to work that way for me. I made the changes here: https://github.com/JoeGruffins/dcrdex/commit/301fc9e9597cf655b21b3031be20246f991b11fa

Reply:
@JoeGruffins this is work great